include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -735,19 +735,20 @@ __parallel_radix_sort_iteration(_ExecutionPolicy&& __exec, ::std::size_t __segme
                                 sycl::event __dependency_event)
 {
     // Injecting ascending / descending status into custom name to prevent clashing kernel names
-    using _Ascending = std::conditional_t<__is_comp_asc, std::true_type, std::false_type>;
+    using _RadixBitsType = ::std::integral_constant<::std::uint32_t, __radix_bits>;
+    using _AscendingType = ::std::bool_constant<__is_comp_asc>;
     using _CustomName = typename __decay_t<_ExecutionPolicy>::kernel_name;
     using _RadixCountKernel = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<
-        __radix_sort_count_kernel, _CustomName, __decay_t<_InRange>, __decay_t<_TmpBuf>, _Ascending>;
+        __radix_sort_count_kernel, _CustomName, _RadixBitsType, _AscendingType, __decay_t<_InRange>, __decay_t<_TmpBuf>>;
     using _RadixLocalScanKernel =
         oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<__radix_sort_scan_kernel_1, _CustomName,
```

Comment:
One open question is why this `kernel_name_generator` uses so few template parameters?

Reply:
Taking a closer look:

I think we need to add `_RadixBitsType` to both  `_RadixLocalScanKernel` and `_RadixGlobalScanKernel`,  as the size of the radix is used in these kernels.  

Also, I think that `__decay_t<_TmpBuf>` needs to be added to  `_RadixGlobalScanKernel`.  `_RadixReorderPeerKernel` and `_RadixReorderKernel`, as they all use it.

Reply:
I'll try to explain.

For `_RadixGlobalScanKernel` there is nothing to add ever (unless the design is changed). It's not kernel name generator, it's kernel name provider, thus we are not generating the name for it, since it's not precompiled kernel.

> Also, I think that `__decay_t<_TmpBuf>` needs to be added to `_RadixGlobalScanKernel`. `_RadixReorderPeerKernel` and `_RadixReorderKernel`, as they all use it

I don't think it's necessary. Temporary buffer doesn't depend on any user provided template argument. It's always a `sycl::buffer<std::uint32_t, ...>`. So adding it doesn't bring value. I am wondering why it's passed as template parameter... Probably, we are lazy...

> I think we need to add `_RadixBitsType` to both `_RadixLocalScanKernel`...

I agree with that.

Reply:
Done.

Reply:
Sure, they are basically in the same situation. But at least for now I suppose we already have guessing that `_RadixBitsType` could be changed. I don't see why it's true for `_TmpBuf`